### DISCUSSION TITLE
Match type narrowing between tooltip and hover

### DISCUSSION THREAD

**Discussion by cdce8p** [microsoft/pylance-release#2683] [Enhancement]
## Environment data

-   Language Server version: 2020.11.0
-   OS and version: macOS
-   Python version: 3.9

## Actual behaviour

The type information differs between tooltip and hover. This issue might be linked to #562.

![Screen Shot 2020-11-11 at 01 07 38](https://user-images.githubusercontent.com/30130371/98748991-1bf79900-23bb-11eb-89e7-ba0501683cd2.png)

## Expected/Suggested behaviour

The hover box should display the narrowed type.

## Code Snippet / Additional information

```py
from __future__ import annotations
from random import random

var: int | None = 42 if random() > 0.5 else None
if isinstance(var, int):
    var
```

**Comment by erictraut** [comment]
The type displayed in the completion suggestion is always the _declared_ type whereas the hover text shows the narrowed type. This is the intended behavior.

**Comment by cdce8p** [comment]
Would it be possible or an idea to show the narrowed type as well? I for one don't usually use hover while writing and I could imaging others doing the same. Especially since the completion list does take the narrowed type into account (which is what I would expect).

The same example just with `var: int | str`:

![Screen Shot 2020-11-11 at 01 40 28](https://user-images.githubusercontent.com/30130371/98750688-e359be80-23be-11eb-9c13-3f422be05042.png)

**Comment by jakebailey** [comment]
Look at old issues; interestingly TS has chosen to do something different here.

![image](https://user-images.githubusercontent.com/5341706/131582475-0f8add38-fabb-4320-8387-502611f28eb9.png)

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
